### PR TITLE
Initial work

### PR DESCRIPTION
**Pull request by JayH5**
What this includes:
- A Python implementation of `gosu`/`su-exec`
- The beginning of a library that makes constructing and running commands easy. (I have several things I want to add here but trying to keep this PR vaguely minimal)

See the README for more info

### CODE REVIEW

**Comment by JayH5** [comment]
Codecov is hung up because I force-pushed a rebase at one point.. Coverage is at ~95% actually.

File: pydexec/command.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class Command(object):
+    def __init__(self, program):
+        self._program = program
+        self._args = []
+        self._user = None
+        self._env = dict(os.environ)
+
+    def args(self, *args):
+        """ Add a list of extra arguments to the command. """
+        self._args += list(args)
```
Comment (jerith) [inline pydexec/command.py]: Would `self._args.extend(args)` work better here?

File: pydexec/command.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    def user(self, user):
+        """
+        Set the user to change to before execution. The ``user`` argument
+        takes the same form as that provided to the ``USER`` command in a
+        Dockerfile, so [<user>][:<group>], where both user and group are
+        optional and user can be a username or UID and group can be a group
+        name or GID.
+        """
+        self._user = User.from_spec(user)
+        return self
```
Comment (jerith) [inline pydexec/command.py:30]: Do we want to allow the user to be set multiple times?

File: pydexec/tests/test_pysu.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    # [ "$(gosu 1000 env | grep '^HOME=')" = 'HOME=/' ]
+    assert_that(pysu_env_home('1000'), Equals(['HOME=/']))
+    # [ "$(gosu 1000:0 env | grep '^HOME=')" = 'HOME=/' ]
+    assert_that(pysu_env_home('1000:0'), Equals(['HOME=/']))
+    # [ "$(gosu 1000:root env | grep '^HOME=')" = 'HOME=/' ]
+    assert_that(pysu_env_home('1000:root'), Equals(['HOME=/']))
+    # [ "$(gosu games env | grep '^HOME=')" = 'HOME=/usr/games' ]
+    assert_that(pysu_env_home('games'), Equals(['HOME=/usr/games']))
+    # [ "$(gosu games:daemon env | grep '^HOME=')" = 'HOME=/usr/games' ]
+    assert_that(pysu_env_home('games:daemon'), Equals(['HOME=/usr/games']))
```
Comment (jerith) [inline pydexec/tests/test_pysu.py]: Can we break this up into smaller tests? We don't necessarily need one per assertion, but some kind of logical grouping would make it easier to reason about.

File: pydexec/tests/test_user.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        """
+        user = User.from_spec(u'')
+
+        assert_that(user.uid, Equals(current_user.uid))
+        assert_that(user.gid, Equals(current_user.gid))
+        assert_that(user.sgroups_user, Equals(current_user.user))
+        assert_that(user.home, Equals(current_user.home))
+
+    def test_invalid_spec(self):
+        """ Passing an invlaid user spec should raise an exception. """
```
Comment (jerith) [inline pydexec/tests/test_user.py]: Tyop: "invlaid"

File: pydexec/command.py (hunk 2 above)
Comment (JayH5) [reply pydexec/command.py:30]: Yes, I think it's reasonable to keep it simple like this. In the future there are going to be other things which can be set multiple times like environment variables.

**Review by jerith — APPROVED** [review]
:+1:

**Pull request merged**